cmd/process-agent/collector.go

```diff
@@ -105,9 +105,43 @@ func (l *Collector) runCheck(c checks.Check, results *api.WeightedQueue) {
 	start := time.Now()
 	// update the last collected timestamp for info
 	updateLastCollectTime(start)
-	messages, err := c.Run(l.cfg, atomic.AddInt32(&l.groupID, 1))
-	if err != nil {
-		log.Errorf("Unable to run check '%s': %s", c.Name(), err)
+
+	if withRealTime, ok := c.(checks.CheckWithRealTime); ok {
+		run, err := withRealTime.RunWithOptions(l.cfg, l.nextGroupID, options)
+		if err != nil {
+			log.Errorf("Unable to run check '%s': %s", c.Name(), err)
+			return
+		}
+		l.messagesToResults(start, withRealTime.Name(), run.Standard, results)
+		l.messagesToResults(start, withRealTime.RealTimeName(), run.RealTime, results)
```

Comment:
Is the same `api.WeightedQueue` passed to both regular and RT checks? https://github.com/DataDog/datadog-agent/pull/8875 created a separate queue for RT checks. I might be missing something but it seems that we're using the same queue for both checks now.

Reply:
Nice catch, yep, this PR predates #8875, I will fix this.

Reply:
done.